formats/cbor/commonMain/src/kotlinx/serialization/cbor/CborEncoder.kt

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright 2017-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.cbor
+
+public interface CborEncoder {
```

Comment:
Yeah, naming is not optimal but for whatever reason when I add a dedicated `CborDecoder` and rename the currently internal `CborEncoder`, which handles low-level stuff, to `LowLevelCborEncoder` the WASI tests glitch out with empty error messages and and empty test report, complaining about failing tests.
So while I agree, I don't even know how to debug this weird behaviour, which, to me, is unintelligible.

Reply:
Managed to split it up by starting from scratch in reverse order. @sandwwraith feel free to use or ignore 82f326fd9b8fc2700583220666e7d6c156195253 as desired.

Reply:
I don't think it matters a lot in this case since you still can invoke `encoder.encodeSerializableValue` using the original interface. But for consistency sake, we can make `CborEncoder` extend `Encoder`.